### ISSUE TITLE
macOS kernel needs to use real application name

### ISSUE THREAD

**Issue by ihnorton** [Slicer/SlicerJupyter#9]
Noting here from quick test: the macOS packaging succeeded last night after the fixes. However, the kernel doesn't run in the packaged Slicer because there is no launcher. This line needs to be changed, because `launcherExecutablePath` is empty string in a packaged build for macOS:

https://github.com/Slicer/SlicerJupyter/blob/3926d2431d366f0e0bff9480c7bff39be9f7b56c/JupyterKernel/qSlicerJupyterKernelModule.cxx#L185

**Comment by ihnorton** [comment]
The kernel ends up trying to run

```
['', '--no-splash', '--python-code', "connection_file=r'/Users/inorton/Library/Jupyter/runtime/kernel-960845b2-8cc5-4253-a6cd-0f2cf97a829d.json'; print('Jupyter connection file: ['+connection_file+']'); slicer.modules.jupyterkernel.startKernel(connection_file);slicer.util.mainWindow().showMinimized()"]
```

which fails.

**Comment by pieper** [comment]
This would be nice to fix anyway - currently Slicer's process name shows up empty in macos utilities like Activity Monitor.  It's never been a real issue before but it's annoying.

![image](https://user-images.githubusercontent.com/126077/43326919-03ffb3e0-9188-11e8-9b5c-cd1a83d48f60.png)

**Comment by lassoan** [comment]
I don't have a Mac, so I cannot fix this. Can you try if replacing ```app->launcherExecutableFilePath()``` by ```slicer.app.applicationFilePath()``` in qSlicerJupyterKernelModule.cxx fixes the problem?

**Comment by jcfr** [comment]
> currently Slicer's process name shows up empty in macos utilities like Activity Monitor

@pieper What this a general comment or is it specific to SlicerJupyer ?

> app->launcherExecutableFilePath() by slicer.app.applicationFilePath() in qSlicerJupyterKernelModule.cxx

:+1:

**Comment by pieper** [comment]
> @pieper What this a general comment or is it specific to SlicerJupyer ?

It looked to me like the same underlying issue - if you fix the application name for SlicerJupyter then it should also fix the issue in the Activity Monitor (and in Force Quit).  But maybe that's not the case if this code is only relying on the launcher.

**Comment by jcfr** [comment]
> fix the issue in the Activity Monito

Problem identified, fix on the way.

**Comment by ihnorton** [comment]
Fixed by #10 (confirmed in 20180729 nightly on Mac)
